store/postgres/src/deployment_store.rs

```diff
@@ -256,13 +256,24 @@ impl DeploymentStore {
         )
     }
 
-    fn check_interface_entity_uniqueness(
+    fn check_interface_entity_uniqueness<'a>(
         &self,
         conn: &PgConnection,
         layout: &Layout,
-        key: &EntityKey,
+        keys: impl Iterator<Item = &'a EntityKey> + Clone,
     ) -> Result<(), StoreError> {
+        let mut keys = keys.peekable();
+
+        // We must receive at least one key
+        let key = match keys.peek() {
+            Some(&key) => key.to_owned(),
+            None => panic!("check_interface_entity_uniqueness received no keys"),
+        };
+
```

Comment:
This function always receives at least one value, so we can expect this to panic only in case of a development mistake, almost like a debug assertion.